service/controller/resource/cloudconfigblob/delete.go

```diff
@@ -41,7 +41,10 @@ func (r *Resource) EnsureDeleted(ctx context.Context, obj interface{}) error {
 	var containerURL azblob.ContainerURL
 	{
 		containerURL, err = r.getContainerURL(ctx, credentialSecret, key.ClusterID(&azureMachinePool), key.StorageAccountName(&azureMachinePool))
-		if err != nil {
+		if IsStorageAccountNotFound(err) {
```

Comment:
This must be handled on creation as well because this can easily happen when creating cluster with defaults (so nodepool gets created together with cluster).